### DISCUSSION TITLE
Theming problem with Niri alongside KDE

### DISCUSSION THREAD

**Discussion by vncnz** [YaLTeR/niri#597] [Q&A]
I hope I chose the right section, I need help because I can't get through it...
I have Niri alongside KDE and I'm having same problem with theming. In KDE settings I have the dark theme selected (breeze-dark) but in Niri every KDE app (Dolphin and kCalc, for example) loads with wrong colors, I'm attaching a clarifying screenshot.
With the light theme is all ok.

Thanks a ton!

![image](https://github.com/user-attachments/assets/e714436c-a67b-43df-88f5-f1f4036f5a8b)

**Comment by YaLTeR** [comment]
Since niri uses the gnome portal, you presumably need to set the gnome dark style property in dconf?

**Comment by vncnz** [comment]
I tried both `dconf write /org/gnome/shell/extensions/user-theme/name "'Breeze-Dark'"` and `gsettings set org.gnome.desktop.interface gtk-theme 'Breeze-Dark'` but without effect, it was already set in GTK too.

Firefox is using the correct theme, getting it from GTK config on the fly, KWrite too. KTorrent uses always a light theme. KCalc and Dolphin are unusable with a mix of light and dark theme.

**Reply by YaLTeR** [reply]
Try: `dconf write /org/gnome/desktop/interface/color-scheme "\"prefer-dark\""`

**Reply by vncnz** [reply]
It was already set, no effect

**Reply by YaLTeR** [reply]
Weird, not sure then. Would have to look at where KDE apps get their theme settings from.

**Reply by YaLTeR** [reply]
You could try adding a Settings portal override to the KDE portal in niri-portals.conf?

**Reply by vncnz** [reply]
I need to install xdg-desktop-portal-kde and add `org.freedesktop.impl.portal.Settings=xdg-desktop-portal-kde.servic`e in `/usr/share/xdg-desktop-portal/niri-portals.conf` ? Sorry but I'm not an expert.

**Reply by YaLTeR** [reply]
If you have KDE then it should already be installed I think? And yeah, something like this if I'm not mistaken. There's also a user home folder override for portals, see `man portals.conf`

**Reply by vncnz** [reply]
The package xdg-desktop-portal-kde wasn't installed, anyway I found that Dolphin and KCalc are written with Qt so I have to fix that style and not GTK one. I don't know how KDE manages this this, I'll try to learn how theming works in the near future (coming weeks, I suppose), I don't have much time for now and I installed Nemo anyway.

Than you very much for Niri, I love it!

**Reply by mlyoung101** [reply]
@vncnz did you find the solution to this? My KDE theme is light on Niri.

**Reply by S0und** [reply]
@mlyoung101 Context: I'm using niri with DMS:  

1. start `qt6ct`, Appearance -> Color scheme: Dank Shell (matugen)(KColorScheme)
2. In case of Dolphin, I had to set the Configure -> Window Color Scheme -> Dank Shell , to have a fully themed file browser.

**Reply by Stranded-Wanderer** [reply]
Did you ever encounter a problem with qt6ct such as this one: 
<img width="500" height="102" alt="image" src="https://github.com/user-attachments/assets/78ec5197-433a-4eaf-977c-430234edb28e" />
